README.md

```diff
@@ -77,71 +77,65 @@ http.request(opts, function (res) {
 * `dist/opentracing-browser.js` - debug version with runtime checks
 
+### Usage with TypeScript
+
+Since the source is written in TypeScript, if you are using TypeScript, you can just `npm install` the package and it will work out of the box.
+This is especially useful for implementators who want to typecheck their implementation with the base interface.
+
 ### Global tracer
 
 The library also provides a global singleton tracer for convenience. This can be set and accessed via the following:
 
 ```javascript
 opentracing.initGlobalTracer(new CustomTracer());
 
-var tracer = opentracing.globalTracer();
+const tracer = opentracing.globalTracer();
 ```
 
 Note: `globalTracer()` returns a wrapper on the actual tracer object. This is done for convenience of use as it ensures that the function will always return a non-null object.  This can be helpful in cases where it is difficult or impossible to know precisely when `initGlobalTracer` is called (for example, when writing a utility library that does not control the initialization process).  For more precise control, individual `Tracer` objects can be used instead of the global tracer.
 
-### Node.js debug version
-
-```javascript
-var opentracing = require('opentracing/debug');
-```
-
-Requiring `opentracing/debug` will include a version of the library with additional runtime checks that are useful for debugging but not desirable for production use.
-
 ## API Documentation
 
-There is a hosted copy of the current generated [ESDoc API Documentation here](https://doc.esdoc.org/github.com/opentracing/opentracing-javascript/).
+There is a hosted copy of the current generated [ESDoc API Documentation here](https://opentracing-javascript.surge.sh).
 
 ## Contributing & developer information
 
 See the [OpenTracing website](http://opentracing.io) for general information on contributing to OpenTracing.
 
-The project is built using a `Makefile`. Run:
+The project is written in TypeScript and built using a npm scripts. Run:
 
-* `make build` creates the compiled, distributable code
-* `make test` runs the tests
+* `npm run build` creates the compiled, distributable JavaScript code in `./lib`
+* `npm run watch` incrementally sompiles on file changes
+* `npm run webpack` creates the bundles for the browser in `./dist`
+* `npm test` runs the tests
+* `npm run typedoc` generates the documentation in `./typedoc`
 
 ## OpenTracing tracer implementations
 
 *This section is intended for developers wishing to* ***implement their own tracers***. *Developers who simply wish to* ***use OpenTracing*** *can safely ignore this information.*
 
 ### Custom tracer implementation
 
-Implementations can subclass `opentracing.Trace`, `opentracing.Span`, and the other API classes to build a OpenTracing tracer.
-
-Due to the dynamic nature of JavaScript, implementations can simply implement classes with the same signatures as the OpenTracing classes and use these directly as well (there's no need to subclass).
```

Comment:
It depends. If you use TypeScript, and type your method to take an opentracing `Span`, it will work as long as the implementation has the same methods and properties. For JS, it doesn't matter of course. But I didn't get why this needs to be highlighted here, that is simply how JS works and should be obvious to any JS dev. I didn't want to confuse people by explaining the exact circumstances, but rather encourage implementations to subclass the opentracing implementations. What do you mean with going through a proxy class?